gatsby-node.js

```diff
@@ -44,7 +45,37 @@ const createHomePage = async (createPage) => {
   })
 }
 
-// TODO: add action for creating section pages
+// TODO: add action for creating section pages]
+const createSections = async (createPage, slug) => {
+  const sectionResp = await axios.get(`${BASE_URL}https://www.thedp.com/section/${slug}.json`)
+  const mostReadDPResp = await axios.get(MOST_READ_DP_API)
+  const topResp = await axios.get(TOP_ARTICLES_API)
+  const { articles } = sectionResp.data
+  const { articles: topArticles } = topResp.data
+
+  let filteredArticles = []
+  if (articles) {
```

Comment:
simplify this code using default params. do const { articles = []} and then you can set const filteredArticles = articles.map(article => ...) always, since articles will always be some sort of array